src/main/java/com/nutomic/syncthingandroid/fragments/DevicesFragment.java

```diff
@@ -54,7 +54,7 @@ public void onActivityCreated(Bundle savedInstanceState) {
 
     private void initAdapter() {
         SyncthingActivity activity = (SyncthingActivity) getActivity();
-        if (activity == null || activity.getApi() == null)
+        if (activity == null || activity.getApi() == null || activity.isFinishing())
```

Comment:
that would look wierd, we have the activity reference just the line above it and test its validity in the same if test